go/consensus/tendermint/apps/registry/transactions.go

```diff
@@ -162,7 +162,15 @@ func (app *registryApplication) registerNode( // nolint: gocyclo
 		)
 		return err
 	}
-	newNode, err := registry.VerifyRegisterNodeArgs(params, app.logger, sigNode, untrustedEntity, ctx.Now(), ctx.IsInitChain(), regRuntimes)
+	regNodes, err := state.Nodes()
```

Comment:
Instead of fetching all nodes I think we should have indices by keys so we would need only a single lookup per registration per key. There can be a lot of nodes so this can otherwise get expensive.

Reply:
That's a good idea, but this can be optimized later (I can do it in next sprint if you want :).  We'd need multiple orders of magnitude more nodes on our testnet than we have now for this to become a bottleneck.

Reply:
Yeah, can you file an issue and assign yourself?

Reply:
Done! #2454 :)